src/lib/OS/PageFrameAllocator.cc

```diff
@@ -0,0 +1,30 @@
+#include "simeng/OS/PageFrameAllocator.hh"
+
+#include <iostream>
+
+#include "simeng/util/Math.hh"
+
+namespace simeng {
+namespace OS {
+
+PageFrameAllocator::PageFrameAllocator(uint64_t maxSize)
+    : maxAllocationSize_(maxSize), sizeLeft_(maxSize) {}
```

Comment:
Could this be moved to header?

Reply:
Given that it isn't an empty constructor shouldn't it be here for styling consistency?

Reply:
and since all other functions are defined are and not in a header file

Reply:
Fair comment, leave as is